### PR TITLE
Feature/move cluster to crd

### PR DESCRIPTION
**Pull request by mrIncompetent**

### CODE REVIEW

File: api/pkg/controller/cluster/controller.go [hunk 1]
```diff
@@ -160,60 +180,82 @@ func (cc *controller) updateNamespace(originalData []byte, ns *v1.Namespace) err
...
 
-func (cc *controller) checkTimeout(c *api.Cluster) (*api.Cluster, error) {
-	now := time.Now()
-	timeSinceCreation := now.Sub(c.Status.LastTransitionTime)
-	if timeSinceCreation > launchTimeout {
-		glog.Infof("Launch timeout for cluster %q after %v", c.Metadata.Name, timeSinceCreation)
-		c.Status.Phase = api.FailedClusterStatusPhase
-		c.Status.LastTransitionTime = now
-		return c, nil
+func (cc *controller) timeoutWorker() {
```
Comment (lbb) [inline api/pkg/controller/cluster/controller.go]: What is this method doing?
Please change this to a more expressive name :). Or in the wort case write a commit.

File: api/pkg/controller/cluster/deleting.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	kubermaticv1 "github.com/kubermatic/kubermatic/api/pkg/crd/kubermatic/v1"
+	kuberneteshelper "github.com/kubermatic/kubermatic/api/pkg/kubernetes"
+
+	"github.com/kubermatic/kubermatic/api/pkg/provider"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (cc *controller) syncDeletingCluster(c *kubermaticv1.Cluster) (changedC *kubermaticv1.Cluster, err error) {
```
Comment (lbb) [inline api/pkg/controller/cluster/deleting.go]: Interesting function 👍 Maybe a bit unintuitive that this is called multiple times, can you somehow comment this please.

**Review by lbb — DISMISSED** [review]

File: api/pkg/controller/cluster/deleting.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go]: Actually this is the normal way how kubermatic handles cluster phases.
I'll comment though if that helps.

**Comment by mrIncompetent** [comment]
@realfake PTAL

**Review by metalmatze — APPROVED** [review]

**Pull request merged**